### ISSUE TITLE
datacenter_id of the assignment is optional from the doc but required from the provider

### ISSUE THREAD

**Issue by luiof** [akamai/terraform-provider-akamai#259]
### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

```
$ terraform -version
Terraform v1.0.9
on darwin_amd64
+ provider registry.terraform.io/akamai/akamai v1.7.1
```

### Affected Resource(s)

akamai_gtm_geomap

### Terraform Configuration Files

Variable exported for the terraform script:
`export TF_VAR_gtm_domain_address="<my GTM address>"`

```
locals {
  geo_map_regions = [
    {
      nickname      = "Southern South America"
      countries = [
        "AR",
        "UY",
        "FK",
        "CL",
        "PY",
        "BO"
      ]
    }, {
      nickname      = "Oceania"
      countries = [
        "TV",
        "FJ",
        "NR",
        "FM",
        "NU",
        "PW",
        "NZ",
        "SB",
        "WF",
        "UM",
        "MH",
        "WS",
        "KI",
        "CC",
        "MP",
        "CK",
        "GU",
        "AS",
        "AU",
        "CX",
        "NC",
        "PF",
        "PG",
        "TK",
        "NF",
        "TO",
        "PN",
        "VU"
      ]
    }, {
      nickname      = "Southern Asia"
      countries = [
        "BD",
        "BT",
        "NP",
        "IN",
        "IO",
        "MV",
        "PK",
        "LK"
      ]
    }, {
      nickname      = "Middle East"
      countries = [
        "IL",
        "AE",
        "AF",
        "JO",
        "UZ",
        "BH",
        "SY",
        "IQ",
        "IR",
        "AM",
        "KW",
        "YE",
        "SA",
        "KZ",
        "QA",
        "LB",
        "TJ",
        "AZ",
        "KG",
        "OM"
      ]
    }, {
      nickname      = "Northern Africa"
      countries = [
        "BF",
        "DJ",
        "BJ",
        "LY",
        "SC",
        "SD",
        "UG",
        "MA",
        "DZ",
        "SL",
        "GA",
        "KE",
        "SN",
        "SO",
        "ML",
        "SS",
        "GH",
        "ST",
        "EG",
        "CF",
        "EH",
        "CG",
        "MR",
        "CI",
        "GM",
        "GN",
        "CM",
        "GQ",
        "ER",
        "ET",
        "GW",
        "TD",
        "CV",
        "TG",
        "NG",
        "TN"
      ]
    }, {
      nickname      = "Southern Africa"
      countries = [
        "CD",
        "KM",
        "RW",
        "LR",
        "TZ",
        "BI",
        "LS",
        "SZ",
        "ZA",
        "MU",
        "MW",
        "MZ",
        "AO",
        "NA",
        "RE",
        "ZM",
        "BV",
        "BW",
        "SH",
        "NE",
        "MG",
        "YT",
        "ZW"
      ]
    }, {
      nickname      = "Southeast Asia"
      countries = [
        "MM",
        "MO",
        "HK",
        "TW",
        "BN",
        "MY",
        "SG",
        "TH",
        "LA",
        "PH",
        "TL",
        "VN",
        "ID",
        "KH"
      ]
    }, {
      nickname      = "Northeast Asia"
      countries = [
        "MN",
        "KP",
        "JP",
        "KR",
        "CN"
      ]
    }, {
      nickname      = "Eastern North America"
      countries = [
        "TT",
        "PR",
        "US/FL",
        "HN",
        "CA/PE",
        "US/NY",
        "DM",
        "DO",
        "HT",
        "US/NC",
        "US/AR",
        "MF",
        "US/NJ",
        "US/VT",
        "US/RI",
        "US/NH",
        "MQ",
        "US/WV",
        "AG",
        "MS",
        "AI",
        "US/KY",
        "VC",
        "CA/QC",
        "VG",
        "VI",
        "US/SC",
        "AW",
        "US/WI",
        "NI",
        "US/OH",
        "US/GA",
        "BB",
        "JM",
        "US/DC",
        "US/DE",
        "CA/NF",
        "BL",
        "CA/NB",
        "BM",
        "BQ",
        "BS",
        "US/LA",
        "US/CT",
        "US/PA",
        "CA/NU",
        "CA/NS",
        "BZ",
        "GD",
        "US/TN",
        "US/IL",
        "US/IN",
        "US/MS",
        "KN",
        "SV",
        "GL",
        "SX",
        "US/MN",
        "US/MO",
        "US/AL",
        "US/VA",
        "GP",
        "GT",
        "KY",
        "CR",
        "TC",
        "US/MA",
        "PA",
        "CU",
        "CW",
        "LC",
        "US/MI",
        "US/MD",
        "PM",
        "US/IA",
        "US/ME",
        "CA/ON"
      ]
    }, {
      nickname      = "Western North America"
      countries = [
        "US/TX",
        "US/NM",
        "US/HI",
        "US/NV",
        "US/WA",
        "US/ND",
        "CA/NT",
        "CA/BC",
        "US/NE",
        "US/AZ",
        "US/OR",
        "CA/YT",
        "US/UT",
        "US/WY",
        "CA/SK",
        "CA/MB",
        "US/CO",
        "MX",
        "US/MT",
        "US/KS",
        "US/AK",
        "US/SD",
        "US/ID",
        "US/OK",
        "US/CA",
        "CA/AB"
      ]
    }, {
      nickname      = "Western Europe"
      countries = [
        "DE",
        "NO",
        "BE",
        "FI",
        "PT",
        "DK",
        "LT",
        "FO",
        "LU",
        "LV",
        "FR",
        "HU",
        "SE",
        "MC",
        "SI",
        "SJ",
        "SK",
        "SM",
        "GB",
        "IE",
        "EE",
        "GG",
        "GI",
        "AD",
        "IM",
        "CH",
        "MT",
        "IS",
        "IT",
        "VA",
        "ES",
        "AT",
        "CZ",
        "JE",
        "PL",
        "LI",
        "NL"
      ]
    }, {
      nickname      = "Eastern Europe"
      countries = [
        "RS",
        "RU",
        "BG",
        "HR",
        "AL",
        "GR",
        "UA",
        "CY",
        "BY",
        "MD",
        "ME",
        "RO",
        "GE",
        "MK",
        "TR",
        "BA"
      ]
    }, {
      nickname      = "Northern South America"
      countries = [
        "BR",
        "GY",
        "PE",
        "CO",
        "GS",
        "EC",
        "GF",
        "SR",
        "VE"
      ]
    }
  ]
}

data "akamai_group" "default" {
  group_name = "<secret>"
  contract_id = "<secret>"
}

# Creating the geo map
resource "akamai_gtm_datacenter" "dc_for_zones" {
  count = length(local.geo_map_regions)
  domain = var.gtm_domain_address
  nickname = "${count.index} added for zones"
  wait_on_complete = false
}

resource "akamai_gtm_geomap" "all_regions" {
  wait_on_complete = false
  # depends_on = [akamai_gtm_datacenter.dc_for_zones]
  domain = var.gtm_domain_address
  name = "13-all-regions"
  default_datacenter {
    datacenter_id = 5400
    nickname = "All Others"
  }

  dynamic assignment {
    for_each = local.geo_map_regions
    content {
      nickname = assignment.value.nickname
      countries = assignment.value.countries
    }
  }
}

```

### Expected Behavior

I am expecting to create the geomap without issues since if I create it manually from the UI of the Akamai panel I have not to insert the datacenter_id and the documentation says it is optional.

### Actual Behavior
I am blocked since it requires the datacenter_id in the assignment section (it is optional from the doc [here](https://registry.terraform.io/providers/akamai/akamai/latest/docs/resources/gtm_geomap#datacenter_id)):
```
│ Error: Missing required argument
│ 
│   on main.tf line 390, in resource "akamai_gtm_geomap" "all_regions":
│  390:     content {
│ 
│ The argument "datacenter_id" is required, but no definition was found.
╵
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:

```
export TF_VAR_gtm_domain_address="<my GTM address>"

terraform init -upgrade
terraform plan
terraform apply -auto-approve
```

**Comment by edglynes** [comment]
The documentation is not correct. The datacenter_id is required.

**Comment by lkowalsk-akamai-com** [comment]
Hi @luiof 
This issue is now fixed with relese v1.9.0 of Akamai Terraform Provider. I'm closing this issue now. Plesae feel free to reaopen the issue if you find it not fixed.
